erpnext/accounts/custom/address.py

```diff
@@ -22,6 +22,18 @@ def validate_reference(self):
 			frappe.throw(_("Address needs to be linked to a Company. Please add a row for Company in the Links table."),
 				title=_("Company Not Linked"))
 
+	def on_update(self):
+		"""
+		After Address is updated, update the related 'Primary Address' on Customer.
+		"""
+		address_display = get_address_display(self.as_dict())
+		filters = { "customer_primary_address": self.name }
+		customers = frappe.db.get_all("Customer", filters=filters, as_list=True)
+		for customer_name in customers:
+			doc_customer = frappe.get_doc("Customer", customer_name[0])
+			doc_customer.primary_address = address_display
+			doc_customer.save()
```

Comment:
use `frappe.db.set_value` or `frappe.db.update`

Reply:
Why do you believe either of those would be an improvement?

Reply:
@PMojito Performance reasons. 

`get_doc` fetches entire document, its child tables etc. Then `save()` goes through all the validations. 

`db.set_value` bypasses all that and updates in a single SQL query.

Reply:
That's good feedback; change complete!